air/src/execution_step/lambda_applier/utils.rs

```diff
@@ -86,6 +87,17 @@ pub(super) fn try_scalar_ref_as_idx(scalar: ScalarRef<'_>) -> LambdaResult<u32>
     }
 }
 
+pub(super) fn try_scalar_ref_as_stream_map_key(scalar: ScalarRef<'_>) -> LambdaResult<StreamMapKey<'static>> {
+    match scalar {
+        ScalarRef::Value(map_accessor) => {
+            let map_accessor = map_accessor.get_result().as_ref().clone();
+            StreamMapKey::from_value(map_accessor.clone())
```

Comment:
Imagine that there is a 'into_owned' function in `StreamMapKey`, then you could do it the following

```rust
            let map_accessor = map_accessor.get_result().as_ref();
            let map_key = StreamMapKey::from_value_ref(map_accessor)
                .ok_or(LambdaError::CanonStreamMapAccessorHasInvalidType { map_accessor: map_accessor.cloned() })?;
            Ok(map_key.into_owned())
```

Here is only one `clone` instead of two in the current variant in case of no error.

Reply:
I can't apply SMK::from_value_ref to an object made in the lifetime of this function.